moment_kinetics/test/calculus_tests.jl

```diff
@@ -29,24 +28,25 @@ function runtests()
 
                 # define inputs needed for the test
                 L = 6.0
-                bc = "periodic"
-                # fd_option and adv_input not actually used so given values unimportant
+                if discretization == "finite_difference"
+                    bc = "periodic"
+                else
+                    bc = "none"
+                end
                 fd_option = ""
-                adv_input = advection_input("default", 1.0, 0.0, 0.0)
                 # create the 'input' struct containing input info needed to create a
                 # coordinate
-                nelement_local = nelement
-				nrank_per_block = 0 # dummy value
-				irank = 0 # dummy value
-				comm = MPI.COMM_NULL # dummy value 
-				input = grid_input("coord", ngrid, nelement,
-                    nelement_local, nrank_per_block, irank, L,
-                    discretization, fd_option, cheb_option, bc, adv_input, comm,
-                    element_spacing_option)
+                input = OptionsDict("coord" => OptionsDict("ngrid"=>ngrid, "nelement"=>nelement,
+                                                           "L"=>L,
+                                                           "discretization"=>discretization,
+                                                           "finite_difference_option"=>fd_option,
+                                                           "cheb_option"=>cheb_option,
+                                                           "bc"=>bc,
+                                                           "element_spacing_option"=>element_spacing_option))
                 # create the coordinate struct 'x'
                 # This test runs effectively in serial, so use `ignore_MPI=true` to avoid
                 # errors due to communicators not being fully set up.
-                x, spectral = define_coordinate(input; ignore_MPI=true)
+                x, spectral = define_coordinate(input, "coord"; ignore_MPI=true, collision_operator_dim=false)
```

Comment:
As an example of my question about this syntax, I don't see why we shouldn't have 

```
input = OptionsDict("name" => "coord", "ngrid" => ... etc)
x, spectral = define_coordinate(input; ignore_MPI=true)
```
Why do we ever need to pass the input for multiple coordinates to define_coordinate?

Reply:
As above, so the code can check that only the right sections exist.

Reply:
Why wouldn't the section exist if we are passing a unique set of coordinate data? I don't see that `set_defaults_and_get_section!()` is appropriate here, but I respect the choice to use existing functions.

Reply:
The idea is to catch mistakes in the input file, e.g. if you made a copy-paste-edit error and created a section called `[vpz]` when you meant `[vz]`. Since (with this PR's code) we know what sections are supposed to exist, we can throw an error and say that `[vpz]` is incorrect. This means that something global needs to know what sections are 'supposed to exist' - that information is stored in `input_dict`, and the fact that the section is 'supposed to exist' is declared by calling `set_defaults_and_check_section!()` (within `define_coordinate()`.

Reply:
I see why this functionality is very very useful when checking input files, but not when checking the functionality for a specific `define_coordinate()` call. To me it just seems that we are reusing an i/o function to manage passing of data internally, with consequences that we need to pass name information twice. If I understand correctly, the check is done on the duplicated name information that is given in almost the same place as the data is defined, not against some global list of allowed coordinates.

Reply:
No, we create a global list, but it's created implicitly (by the `set_defaults_and_check_section!()` calls), so that we don't have to maintain a global list separately, which is required to be consistent and would cause bugs when it isn't.

The calls to `define_coordinate()` in `mk_input()` add the `r`, `z`, `vperp`, `vpa`, `vzeta`, `vr`, `vz` sections to `input_dict["_section_check_store"]`. When all the input sections have been read, `input_dict["_section_check_store"]` contains the list of all the sections, and then we check in `check_sections!()` that the input didn't have any extra sections that weren't read.

So I think the current version is useful and necessary (to check that there are no extra sections that would be silently ignored). We could have an extra version of `define_coordinate()` that just took the input for a single coordinate with no name, and I get the point it would simplify some of the test code by not having to create a subsection to pass input, but then the tests wouldn't be using the same function as the main code, which would probably be OK but contains some risk of causing problems. I don't think getting rid of a few `"z" => OptionsDict(` characters is worth that (admittedly small) risk, and the question for a new user of 'why do calls to `define_coordinate()` in the tests look different from calls to `define_coordinate()` in `mk_input()?`.

Reply:
Although if something like
```julia
function define_test_coordinate(input_dict)
    input_dict = deepcopy(input_dict)
    name = pop!(input_dict, "name")
    return define_coordinate(OptionsDict(name => input_dict), name)
```
would make you happy, we can add that. Could be worth doing, as it would simplify the syntax in some tests.

We could even have a kwargs version so we don't have to define an `OptionsDict` like
```julia
function define_test_coordinate(; kwargs...)
    return define_test_coordinate(OptionsDict(String(k) => v for (k,v) in kwargs)
end
```

Reply:
Do you ever want distributed-MPI in the test coordinates? If so we'd need to pass some of the keyword arguments through `define_test_coordinate()`.

Reply:
So far I haven't needed distributed MPI in the tests. Maybe that would be desirable in the future -- but I think the bigger challenge connected to that is making a distributed LU decomposition and distribution LU solve so that the weak-form calculus can actually use distributed MPI.

Reply:
I've added `define_test_coordinate()` now, so I think this is resolved.